### PR TITLE
feat(fn): introduce dialog, message-box, link, title [ci visual]

### PR DESCRIPTION
**Pull request by droshev**
part of https://github.com/SAP/fundamental-styles/issues/2928

Introduce FN dialog, message box, title and link

### CODE REVIEW

File: src/fn/fn-title.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  &--h2 {
+    font-size: 2rem;
+  }
+
+  &--h3 {
+    font-size: 1.5rem;
+  }
+
+  &--h4 {
+    font-size: 1.25;
```
Comment (InnaAtanasova) [inline src/fn/fn-title.scss:30]: missing rem

File: src/fn/fn-message-box.scss [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+$fn-message-box-icon-text-spacing: 0.5rem;
+$fn-message-box-title-icon-text-size: 1rem;
+
+$message-box-states: (
+  "confirmation": ("iconColor": $fn-color-grey-500),
+  "error": ("iconColor": $fn-color-red-500),
+  "success": ("iconColor": $fn-color-green-500),
+  "warning": ("iconColor": $fn-color-orange-400),
+  "information": ("iconColor": $fn-color-blue-500)
```
Comment (InnaAtanasova) [inline src/fn/fn-message-box.scss:23]: we should stick with the naming used in the design: positive, critical, negative...

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-12-01 at 11 22 59 AM

The link should probably get the appropriate cursor and underline on hover and active

File: src/fn/fn-message-box.scss [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  .fn-message-box+(--active)
+    .fn-message-box__content
+      .fn-message-box__header
+      .fn-message-box__body+(--no-vertical-padding)
+      .fn-message-box__footer
+*/
+
+$block: #{$fn-namespace}-message-box;
+
+$fn-message-box-icon-text-spacing: 0.5rem;
```
Comment (InnaAtanasova) [inline src/fn/fn-message-box.scss:15]: are you sure about this value?

**Comment by InnaAtanasova** [comment]
In my opinion it makes more sense to use the text from fd-title as it's more clear to the user:


Screen Shot 2021-12-01 at 11 34 22 AM

instead of:
Screen Shot 2021-12-01 at 11 34 31 AM

I'd even use
```
Title Heading 1
Title Heading 2
Title Heading 3
...
```

But it's a personal opinion so it's not part of the requested changes :)

File: src/fn/fn-link.scss [hunk 4]
```diff
@@ -0,0 +1,11 @@
+@import "./fn-settings";
+
+$block: #{$fn-namespace}-link;
```
Comment (InnaAtanasova) [inline src/fn/fn-link.scss:3]: if this is a separate component we should have a page for it, with all states that can be applied: hover, active, disabled... so we can detect future changes. The cursor is not set and the underline is missing.

File: src/fn/_fn-mixins.scss [hunk 5]
```diff
@@ -1,8 +1,8 @@
 @mixin fn-reset {
   font-family: "72", "72full", Arial, Helvetica, sans-serif;
   font-size: 0.875rem;
-  line-height: 1.5rem;
-  color: #223548;
+  line-height: 1.4;
```
[the remark this answers is not in the document]
Comment (InnaAtanasova) [reply src/fn/_fn-mixins.scss:4]: in most cases I see in Figma the text with 14px has line-height of 22px

File: src/fn/fn-link.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+.#{$block} {
+  @include fn-reset();
+
+  color: $fn-color-blue-500;
+  display: inline-block;
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-link.scss:9]: remove inline, add underline on hover, cursor update

File: src/fn/_fn-mixins.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/_fn-mixins.scss:4]: reverted

File: src/fn/fn-dialog.scss [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    background-color: $fn-color-white;
+    padding: 1.5rem 0 0.75rem 0;
+  }
+
+  &__body {
+    @include fn-reset();
+
+    @extend %dialog-body;
+
+    padding: 1.5rem;
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-dialog.scss:40]: you are right

File: src/fn/fn-dialog.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  &__header {
+    padding: 0 1.5rem;
+  }
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-dialog.scss:45]: removed

File: src/fn/fn-link.scss [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  line-height: inherit;
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-link.scss:10]: removed

File: src/fn/fn-link.scss (hunk 4 above)
Comment (droshev) [reply src/fn/fn-link.scss:3]: i will skip the doc page for now. I will add the rest

File: src/fn/fn-link.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-link.scss:9]: done

File: src/fn/fn-link.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-link.scss:10]: removed

File: src/fn/fn-link.scss (hunk 4 above)
Comment (droshev) [reply src/fn/fn-link.scss:3]: done

File: src/fn/fn-message-box.scss (hunk 2 above)
Comment (droshev) [reply src/fn/fn-message-box.scss:23]: updated

File: src/fn/fn-message-box.scss [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+  &__header {
+    padding: 0 1.5rem;
+  }
+
+  &__footer {
+    padding: 0.375rem 1.5rem;
+    display: flex;
+    justify-content: flex-end;
+    gap: 0.375rem;
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-message-box.scss:65]: actually i removed the code from the dialog

File: src/fn/fn-message-box.scss [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  @each $set-name, $state-set in $message-box-states {
+    &--#{$set-name} {
+
+      .#{$block}__content {
+        border: 2px solid map_get($state-set, "iconColor");
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-message-box.scss:72]: updated

File: src/fn/fn-message-box.scss (hunk 3 above)
Comment (droshev) [reply src/fn/fn-message-box.scss:15]: changed

File: src/fn/fn-title.scss (hunk 1 above)
Comment (droshev) [reply src/fn/fn-title.scss:30]: done

File: src/fn/fn-title.scss [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+@import "./fn-mixins";
+@import "./dialog-placeholders";
+
+$block: #{$fn-namespace}-title;
+$fn-dialog-title-font-size: 1.125rem;
+
+.#{$block} {
+  @include fn-reset();
+
+  @extend %dialog-title;
```
[the remark this answers is not in the document]
Comment (droshev) [reply src/fn/fn-title.scss:11]: done

File: src/fn/fn-link.scss [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    text-decoration: underline;
+  }
+
+  @include fn-active() {
+    text-decoration: underline;
+  }
+
+  @include fn-disabled() {
+    pointer-events: none;
+    opacity: 0.4;
```
Comment (InnaAtanasova) [inline src/fn/fn-link.scss:25]: There's a variable `$fn-disabled-opacity` in the `_fn-new-variables.scss` file

File: src/fn/fn-link.scss [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  color: $fn-color-blue-500;
+  text-decoration: none;
+
+  @include fn-hover() {
+    text-decoration: underline;
+  }
+
+  @include fn-focus() {
+    text-decoration: underline;
+  }
```
Comment (InnaAtanasova) [inline src/fn/fn-link.scss:17]: Your focus is not correct. You need to write code for the following cases:
(You can check Button implementation)
```
@include fn-is-focus() { ... }
@include fn-not-focus-visible() { ... }
@include fn-focus-visible() { ... }
```
Screen Shot 2021-12-10 at 11 08 03 AM

Focus state does not have underline.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/fn/fn-title.scss [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+);
+
+.#{$block} {
+  @include fn-reset();
+  @include fn-ellipsis();
+
+  font-size: 1.125rem;
+  font-weight: bold;
+  color: $fn-color-grey-800;
+  line-height: 1.4;
```
Comment (droshev) [inline src/fn/fn-title.scss:22]: @InnaAtanasova  in `rem` doesn't scale and it cuts the titles

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**